### PR TITLE
DatagovukPlugin: add before_index method

### PR DESCRIPTION
**Pull request by risicle**
https://trello.com/c/r2q5qNr8/

Solr6+ limits fields of type `StrField` to a length of ~32k, but ckan's `schema.xml` has a catchall field which will add unmentioned fields as `StrFields`. This means we can only afford to have strings greater than 32k under keys we're pretty sure the schema has `TextFields` for. Take this list of "safe" field name patterns from the `ckan28` schema as @ https://github.com/alphagov/govuk-puppet/blob/a927fcdf8210bfb4f810ca619a0c4df9fcde425a/modules/govuk_solr6/files/ckan28.schema.xml

Also bump `ckanext-spatial` version to that matching `master`.

### CODE REVIEW

**Comment by risicle** [comment]
This is causing problems because it truncates the `data_dict` field, which is a json string that ckan expects to be able to deserialize, and obviously it now can't.

**Comment by risicle** [comment]
Turns out `data_dict` and `validated_data_dict` are safe to be >32k because they aren't actually _indexed_, being mentioned in the `schema.xml` with `indexed="false"`.

Reduced truncation to 15k in the end to allow for 2 such "large" fields existing in the same complex object. This is obviously not a complete solution for this, but works for now. In future we may have to do something clever involving performing the json-squashing of complex values ourselves in `before_index` so we can know what the final value's length is going to be and adjust truncation appropriately.

File: ckanext/datagovuk/schema.py [hunk 1]
```diff
@@ -53,3 +58,59 @@ def user_edit_form_schema():
...
+        }
+    elif isinstance(value, collections_abc.Sequence):
+        # already eliminated possibility of this being a string
+        return [_trim_strings_for_index_inner(v, limit) for v in value]
+
+    return value
+
+def trim_strings_for_index(
+    data_dict,
+    limit=15000,  # approx half solr6's string-field length limit
```
[the remark this answers is not in the document]
Comment (risicle) [reply ckanext/datagovuk/schema.py:103]: Yeah, do we actually have a way of changing configuration without having to do a new code release anyway?

[the remark this answers is not in the document]
Comment (kentsanggds) [reply ckanext/datagovuk/schema.py]: yes, if the config was updated in puppet that should trigger a reload of the app as it detects changes in the ckan config

**Pull request merged**